src/Account/Infrastructure/CustomerRegisterFactory.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * Copyright © OXID eSales AG. All rights reserved.
+ * See LICENSE file for license details.
+ */
+
+declare(strict_types=1);
+
+namespace OxidEsales\GraphQL\Account\Account\Infrastructure;
+
+use DateTimeInterface;
+use OxidEsales\Eshop\Application\Model\User;
+use OxidEsales\GraphQL\Account\Account\DataType\Customer;
+use OxidEsales\GraphQL\Account\Account\Exception\CustomerExists;
+use OxidEsales\GraphQL\Account\Account\Exception\Password;
+use OxidEsales\GraphQL\Account\NewsletterStatus\Exception\EmailEmpty;
+use OxidEsales\GraphQL\Account\NewsletterStatus\Exception\InvalidEmail;
+use OxidEsales\GraphQL\Base\Service\Authentication;
+use OxidEsales\GraphQL\Base\Service\Legacy;
+
+final class CustomerRegisterFactory
+{
+    /** @var Authentication */
+    private $authentication;
+
+    /** @var Repository */
+    private $repository;
+
+    /** @var Legacy */
+    private $legacyService;
+
+    public function __construct(
+        Authentication $authentication,
+        Repository $repository,
+        Legacy $legacyService
+    ) {
+        $this->authentication = $authentication;
+        $this->repository     = $repository;
+        $this->legacyService  = $legacyService;
+    }
+
+    public function createValidCustomer(string $email, string $password, ?DateTimeInterface $birthdate): Customer
+    {
+        if (!strlen($email)) {
+            throw new EmailEmpty();
+        }
+
+        if (!$this->legacyService->isValidEmail($email)) {
+            throw new InvalidEmail();
+        }
```

Comment:
I would prefer to have on `InvalidEmail` exception only and give two different reasons by named constructor.

`InvalidEmail::byEmptyString` / `InvalidEmail::byString($email)` for example